data/contract_docs/logs.md

```diff
@@ -0,0 +1,97 @@
+# Logs
+
+Status: EARLY Draft/Proposal
+
+## LogLines
+
+### Properties and field requirements
+
+- **Time field** - _required_
+  - There must be at least one non nullable time field
+  - If there are multiple time fields present, following will decide the priority
+    - First matching time field with name `timestamp`
+    - or first matching time field with name `ts`
+    - or first matching time field
+- **Message field** - _required_
+  - There must be at lease one non nullable string field must present
+  - If more than one string fields found, the following will decide the priority
+    - First matching string field with name `body`
+    - or first matching string field with name `message`
+    - or first matching string field
+- **Severity field** - _optional_
+  - This is optional field
+  - Level/Severity of the log line can be represented with this field.
+  - This have to be a string field. (either nullable or non-nullable string field)
+  - Log field will be decided in the following order
+    - First matching string field with name `severity`
+    - or first matching string field with name `level`
+  - If no level field found, consumers/client will decide the log level. Example: logs panels will try to parse the message field and determine the log level
+  - Log level can be one of the values specified in the docs [here](https://grafana.com/docs/grafana/latest/explore/logs-integration/)
+- **ID field** - _optional_
+  - This optional field
+  - Unique identified of the log line
+  - This have to be a string field. (either nullable or non-nullable string field)
+  - Id field will be decided in the following order
+    - First matching string field with name `id`
+    - or first matching string field with name `guid`
+  - If no id field found, consumers/client will decide the id field as required.
+- **Attributes field** - _optional_ / **Labels field**
+  - This is an optional field
+  - This field represent additional attributes of the log line. This is also known as labels field.
+  - Field type must be json raw message type. Example value: `{}`, `{"hello":"world", "foo": 123.45, "bar" :["yellow","red"], "baz" : { "name": "alice" }}`
+    - Should not be empty string.
+    - Value should be represented with `Record<string|number,any>` type in javascript.
```

Comment:
an open question is.. Grafana currently defines `Labels` here: https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/data.ts#L140-L142

as a string-string object,

so..hmm.. maybe, if we go with a new thing (allowing `any`), maybe we should call it something else, not Labels